# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLSync.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLSync.h
--- a/dom/canvas/WebGLSync.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLSync.h	Wed Jan 08 22:19:14 2020 +0000
@@ -15,8 +15,7 @@
 class AvailabilityRunnable;
 }  // namespace webgl
 
-class WebGLSync final : public nsWrapperCache,
-                        public WebGLRefCountedObject<WebGLSync>,
+class WebGLSync final : public WebGLRefCountedObject<WebGLSync>,
                         public LinkedListElement<WebGLSync> {
   friend class WebGL2Context;
   friend class webgl::AvailabilityRunnable;
@@ -29,13 +28,8 @@
   WebGLSync(WebGLContext* webgl, GLenum condition, GLbitfield flags);
 
   void Delete();
-  WebGLContext* GetParentObject() const;
 
-  virtual JSObject* WrapObject(JSContext* cx,
-                               JS::Handle<JSObject*> givenProto) override;
-
-  NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING(WebGLSync)
-  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_NATIVE_CLASS(WebGLSync)
+  NS_INLINE_DECL_REFCOUNTING(WebGLSync)
 
   void MarkSignaled() const;
 